src/ja/worker/config.py

```diff
@@ -10,27 +9,73 @@ class WorkerConfig(Config):
     Config for the worker client.
     """
 
+    def __init__(self, uid: str, ssh_config: SSHConfig, resource_allocation: ResourceAllocation):
+        """!
+        creates resource allocation instance
+        """
+        if uid is None:
+            raise ValueError("missing uid")
+        if ssh_config is None:
+            raise ValueError("missing ssh config")
+        if resource_allocation is None:
+            raise ValueError("missing resources")
```

Comment:
As I've said before, please write more detailed error messages.
(Although I'm not sure you even need them in this case.)